### PR TITLE
telemetry doc.

### PR DESCRIPTION
**Pull request by chieftn**

### CODE REVIEW

File: documents/telemetry.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+Application Insights provides immediate benefits to track solution health and performance.  Additionally, the client SDK provides several performance optimizations, namely batched submits to the telemetry endpoint. However, the low barrier to entry poses some limitations.  The Application Insights SDK downloads from a distinct domain, typically https://az416426.vo.msecnd.net/scripts/a/ai.0.js.  And telemetry is submitted to other domains (e.g. dc.services.visualstudio.com).  This configuration raises the potential for Ad Blockers to limit telemetry collection.  A review of two prominent ad blockers yielded the following results:
+
+1. By default, sample ad blockers did not block Application Insights.
+2. Sample ad blockers could be configured by user to block telemetry collection.
+
+Full details regarding Application Insights enpoints reside [here](https://docs.microsoft.com/en-us/azure/application-insights/app-insights-ip-addresses).  Application Insights' discussion of ad blockers resides [here](https://github.com/Microsoft/ApplicationInsights-JS/issues/411)
+
+Develop teams may wish to weigh the advantages and disadvantages before committing to a telemetry solution.
```
Comment (montgomp) [inline documents/telemetry.md]: Should this be "Development teams"

File: documents/telemetry.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+## Application Insights: Advantages and Limitations
+Application Insights provides immediate benefits to track solution health and performance.  Additionally, the client SDK provides several performance optimizations, namely batched submits to the telemetry endpoint. However, the low barrier to entry poses some limitations.  The Application Insights SDK downloads from a distinct domain, typically https://az416426.vo.msecnd.net/scripts/a/ai.0.js.  And telemetry is submitted to other domains (e.g. dc.services.visualstudio.com).  This configuration raises the potential for Ad Blockers to limit telemetry collection.  A review of two prominent ad blockers yielded the following results:
+
+1. By default, sample ad blockers did not block Application Insights.
+2. Sample ad blockers could be configured by user to block telemetry collection.
+
+Full details regarding Application Insights enpoints reside [here](https://docs.microsoft.com/en-us/azure/application-insights/app-insights-ip-addresses).  Application Insights' discussion of ad blockers resides [here](https://github.com/Microsoft/ApplicationInsights-JS/issues/411)
```
Comment (montgomp) [inline documents/telemetry.md]: typo "endpoints"

Comment (chieftn) [reply documents/telemetry.md]: idunno --maybe i guess :)

File: documents/telemetry.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+4. `LogTrace`:  Logs a trace message with associated severity level.
+
+All methods offer an optional parameter (`properties`) to specify custom key-value string tuples.  LogEvent, LogException, LogPageView also offer the `measurements` parameter to specify custom key-value number tuples. 
+
+Examples reside in the following locations:
+1. `router.tsx`: Demonstrates how to log a page view.
+2. `themer.tsx`: Demonstrates how to log an event with custom properties.
+3. `fetchAzureSubscriptionsSaga.ts`: Demonstrates how to log an exception within a saga.
+
+Notably, the TelemetryService.ts exists to add indirection between telemetry endpoint and consuming code.  It does not implement all Application Insights SDK methods to avoid potentially leaky future abstractions.  Nonetheless, method parameters favor the SDK's approach, projecting its open schema for custom properties and measurements.  Teams with custom telemetry endpoints and metrics may require more specialized adaptations.
```
Comment (iluican) [inline documents/telemetry.md]: I'd change "to add indirection" with "as a layer"

**Pull request merged**